x/evm/keeper/chainKeeper.go

```diff
@@ -206,7 +208,7 @@ func (k chainKeeper) getTokenAddress(ctx sdk.Context, assetName string, details
 }
 
 // GetBurnerAddressAndSalt calculates a burner address and the corresponding salt for the given token address, recipient and axelar gateway address
-func (k chainKeeper) GetBurnerAddressAndSalt(ctx sdk.Context, tokenAddr types.Address, recipient string, gatewayAddr common.Address) (common.Address, common.Hash, error) {
+func (k chainKeeper) GetBurnerAddressAndSalt(ctx sdk.Context, tokenAddr types.Address, recipient string, gatewayAddr common.Address, nonce []byte) (common.Address, common.Hash, error) {
```

Comment:
That is how I was doing it initially, but then I changed it to this so that I could create unit tests more easily. Though I cannot remember the exact reason anymore.

Reply:
refactored the code to remove this parameter from this function